core/src/main/resources/wasmruntime.js

```diff
@@ -389,6 +389,12 @@ const bytecoder = {
                 }
                 return 0;
             },
+            Z$matches$Ljava$lang$String$: function (self, regex) {
+                var match = RegExp(regex.nativeObject).exec(self.nativeObject);
```

Comment:
Looks good! But we have to be careful here. The wasm backend handles nativeObject references differently to the JS backend. Wasm can only access such instances by calling `bytecoder.getNativeObject` or `bytecoder.setNativeObject`

Reply:
It should be `RegExp(bytecoder.getNativeObject(regex)).exec(bytecoder.getNativeObject(self))`